src/Http/Routing/test/UnitTests/Matching/RouteMatcherConformanceTest.cs

```diff
@@ -1,10 +1,46 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Threading.Tasks;
+using Xunit;
+
 namespace Microsoft.AspNetCore.Routing.Matching
 {
     public class RouteMatcherConformanceTest : FullFeaturedMatcherConformanceTest
     {
+        // https://github.com/dotnet/aspnetcore/issues/18677
+        //
+        [Theory]
+        [InlineData("/middleware", 1)]
+        [InlineData("/middleware/test", 1)]
+        [InlineData("/middleware/test1/test2", 1)]
+        [InlineData("/bill/boga", 0)]
+        public virtual async Task Match_Regression_1867(string path, int endpointIndex)
```

Comment:
This test now needs to be special cased across different matchers, so it has to be duplicated.